core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -83,13 +83,15 @@ private[spark] class MetricsSystem private (
 
   // Treat MetricsServlet as a special sink as it should be exposed to add handlers to web ui
   private var metricsServlet: Option[MetricsServlet] = None
+  private var prometheusServlet: Option[PrometheusServlet] = None
 
   /**
    * Get any UI handlers used by this metrics system; can only be called after start().
    */
   def getServletHandlers: Array[ServletContextHandler] = {
     require(running, "Can only call getServletHandlers on a running MetricsSystem")
-    metricsServlet.map(_.getHandlers(conf)).getOrElse(Array())
+    metricsServlet.map(_.getHandlers(conf)).getOrElse(Array()) ++
```

Comment:
Can you just write `(metricsServlet ++ prometheusServlet).map...`?

Reply:
Ur, for now, the first one is `MetricsServlet` and the other is `PrometheusServlet`. `getHandlers` is not inherited from `Sink`.

Reply:
Oh I see.